src/utils/api.js

```diff
@@ -31,6 +33,98 @@ export const formatActions = (actionsByDayObject) => {
   });
 };
 
+// Takes array of action objects and returns an object with {key: value} pairs of {hourOfDay: #-of-actions}
+export const getActionsByTimeOfDay = (actions) => {
+  const actionsByTimeOfDay = {
+    'Late night': 0,
+    'Early morning': 0,
+    Morning: 0,
+    Afternoon: 0,
+    Evening: 0,
+    Night: 0,
+  };
+  actions.forEach((action) => {
```

Comment:
Same as above. Default to ignoring and showing/logging a warning.

Reply:
I'm assuming you mean a `switch` statement. if so, might be a bit weird because `switch` doesn't work well with comparison operators. we can't do `switch (actionHourOfDay) {case >=0...}`; instead it would have to be `switch (true) {case (actionHourOfDay >= 0 && actionHourOfDay <= 4)...}`. let me know if I'm missing anything!

Reply:
Here I'm not referring to the `switch` statement, but to the idea of having an `else` that is not related to an actual case, but otherwise "handles" the error (of not having a correct `actionHourOfDay`). It's just a sanity check.